test/verify.tests.js

```diff
@@ -30,39 +30,52 @@ describe('verify', function() {
     });
   });
 
-  it('should be able to validate unsigned token', function (done) {
+  it('should not be able to verify unsigned token', function () {
     var header = { alg: 'none' };
     var payload = { iat: Math.floor(Date.now() / 1000 ) };
 
     var signed = jws.sign({
       header: header,
       payload: payload,
-      secret: priv,
+      secret: 'secret',
       encoding: 'utf8'
     });
 
-    jwt.verify(signed, null, {typ: 'JWT'}, function(err, p) {
+    expect(function () {
+      jwt.verify(signed, 'secret', {typ: 'JWT'});
```

Comment:
This case would already return an error before your changes, as you're passing in a secret and we do this check:
```
    if (!hasSignature && secretOrPublicKey){
      return done(new JsonWebTokenError('jwt signature is required'));
    }
```
We should have a test for both cases where secret is provided, or is null.